src/common/cockpitconf.c

```diff
@@ -115,188 +175,202 @@ cockpit_conf_init (void)
+
+  free (section_lower);
+  free (field_lower);
+  return e;
 }
 
-const gchar **
-cockpit_conf_strv (const gchar *section,
-                   const gchar *field,
-                   gchar delimiter)
+const char *
+cockpit_conf_string (const char *section,
+                     const char *field)
 {
-  const gchar **value = NULL;
-  gchar *key = NULL;
-  gchar delm[2] = {delimiter, '\0'};
+  const Entry *e = cockpit_conf_lookup (section, field);
+  return e ? e->value : NULL;
+}
 
-  g_return_val_if_fail (section != NULL, NULL);
-  g_return_val_if_fail (field != NULL, NULL);
+bool
+cockpit_conf_bool (const char *section,
+                   const char *field,
+                   bool defawlt)
+{
+  const char *value = cockpit_conf_string (section, field);
+  if (value)
+    return strcasecmp (value, "yes") == 0 || strcasecmp (value, "true") == 0 || strcmp (value, "1") == 0;
+  return defawlt;
+}
 
-  ensure_cockpit_conf ();
-  key = g_strjoin (delm, section, field, NULL);
+static const char **
+get_strv (Entry *entry, char delimiter)
+{
+  char *stripped, *token;
+  char delimiters[2] = "x";
+  unsigned i;
+
+  if (!entry->value)
+    return NULL;
 
-  if (!g_hash_table_contains (cached_strvs, key))
+  if (entry->strv_cache)
     {
-      value = build_strv (section, field, delm);
-      if (value)
-        g_hash_table_insert (cached_strvs, g_strdup (key), value);
+      if (delimiter != entry->strv_delimiter)
+        errx (1, "cockpitconf: Looking up strv with different delimiters is not supported");
+      return (const char**) entry->strv_cache;
     }
-  else
+
+  /* strip off leading whitespace */
+  for (stripped = entry->value; *stripped && isspace (*stripped); ++stripped)
+  /* strip off trailing whitespace */
+  stripped = strdup (stripped);
+  for (char *c = stripped + strlen (stripped) - 1; c >= stripped && isspace (*c); --c)
+    *c = '\0';
+
+  delimiters[0] = delimiter;
+
+  for (i = 0, token = strtok (stripped, delimiters);
+       token;
+       i++, token = strtok (NULL, delimiters))
     {
-      value = g_hash_table_lookup (cached_strvs, key);
+      entry->strv_cache = reallocarr (entry->strv_cache, i + 2, sizeof (char*));
+      entry->strv_cache[i] = strdup (token);
     }
+  entry->strv_cache[i] = NULL;
 
-  g_free (key);
-  return value;
+  entry->strv_delimiter = delimiter;
+  return (const char**) entry->strv_cache;
+}
+
+const char **
+cockpit_conf_strv (const char *section,
+                   const char *field,
+                   char delimiter)
+{
+  Entry *e = cockpit_conf_lookup (section, field);
+  return e ? get_strv (e, delimiter) : NULL;
 }
 
-guint
-cockpit_conf_guint (const gchar *section,
-                    const gchar *field,
-                    guint default_value,
-                    guint64 max,
-                    guint64 min)
+unsigned
+cockpit_conf_uint (const char *section,
+                   const char *field,
+                   unsigned default_value,
+                   unsigned max,
```

Comment:
`unsigned` is only 32 bits. Is that ok?

Reply:
For the limits you mean? I didn't understand why they previously were 64 bit types while the actual value was a `guint`  aka. `unsigned`, i. e 32 bit as well. That said, the only use case for this is setting the timeout option, which is bound between 1 and 900. So it doesn't matter indeed.